### ISSUE TITLE
Waterfox Classic: 10 second delay opening image

### ISSUE THREAD

**Issue by GTR8000** [qsniyg/maxurl#1059]
This affects recent versions of Waterfox up to the current 2022.08 (released a week ago), and possibly some older versions of Firefox.

**In a nutshell, when using the script there is a 10 second delay in opening images after hitting the popup trigger key.**

This behavior only seems to happen with the first image loaded on a webpage. Subsequent popup trigger key presses on the same page open the image within the normal timeframe, i.e. if scrolling down Instagram. Once you change URL's, the behavior starts again with the first image on the new page. I couldn't find anything obvious in the console log while looking for clues.

I've experienced this issue with every website I've used the script with, not specific to any one site.

I also had to completely disable Redirection, as that was causing issues with all images in the browser, even when not specifically using the Image Max URL script.

I've already sent @qsniyg more detail via email, but wanted to create a ticket here to document the issue.

**Comment by qsniyg** [comment]
I can finally replicate the issue. It only happens under the userscript version, not the addon version.

**Comment by qsniyg** [comment]
Setting "Allow showing partially loaded" to "Media" removes the issue for images on my end. Of course, this is not a proper fix, but it should help to find the actual root of the problem.

**Comment by GTR8000** [comment]
No change in behavior on my end. Actually, since I have the 'Popup action' set to 'New background tab', 'Allow showing partially loaded' wasn't even an option; I had to first revert back to "Popup". Still no joy, 10 second delay on a newly opened URL in effect. still running Waterfox Classic 2022.08 as in the original bug report.

**Comment by GTR8000** [comment]
I'm closing this ticket out, as I've changed browsers and this is no longer an issue for me.

**Comment by qsniyg** [comment]
Interesting, sorry that I couldn't replicate what you were experiencing!

**Comment by GTR8000** [comment]
No worries, it actually worked out for the best. I should've moved on from that old browser a long time ago, but there were some legacy extensions that I didn't want to let go of which wouldn't run in modern versions of Firefox/Waterfox. I finally got so frustrated with this bug taking forever to load images, that last week I spent a few hours finding modern replacement extensions and migrating to the latest release of Waterfox (based on Firefox 102 ESR), so I'm in much better shape now!
